platform/ios/src/MGLMapView.mm

```diff
@@ -4670,15 +4670,25 @@ - (void)applyIconIdentifier:(NSString *)iconIdentifier toAnnotationsWithImageReu
 
 #pragma mark - User Location -
 
+- (void)setLocationManager:(id<MGLLocationManager>)locationManager
+{
+    _locationManager = locationManager;
+    _locationManager.delegate = self;
+    [self validateLocationServices];
+}
+
 - (void)validateLocationServices
 {
     BOOL shouldEnableLocationServices = self.showsUserLocation && !self.dormant;
 
-    if (shouldEnableLocationServices && ! self.locationManager)
+    if (shouldEnableLocationServices)
     {
-        self.locationManager = [[CLLocationManager alloc] init];
+        // If no custom location manager is provided will use the internal implementation.
+        if (!self.locationManager) {
+            self.locationManager = [[MGLCLLocationManager alloc] init];
```

Comment:
This will call `-validateLocationServices` again, won't it?

Reply:
It won't. https://github.com/mapbox/mapbox-gl-native/pull/12013/files#diff-20502b6bc02986a4398d0da84b2d4b94R4673

Reply:
Perfect :) Looks like I was commenting on an earlier version.